### PR TITLE
Stricter parser for Julia

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/inlineasm/AsmParser.cpp [hunk 1]
```diff
@@ -119,12 +120,16 @@ assembly::Statement Parser::parseStatement()
...
 			// label
+			if (m_julia)
+				fatalParserError("Labels are not supported in Julia.");
 			Label label = createWithLocation<Label>(identifier.location);
 			label.name = identifier.name;
 			return label;
 		}
 	}
 	default:
+		if (m_julia)
```
Comment (axic) [inline libsolidity/inlineasm/AsmParser.cpp:131]: This default case in non-Julia mode results in the literal pushes.

File: libsolidity/inlineasm/AsmParser.cpp [hunk 2]
```diff
@@ -84,14 +84,15 @@ assembly::Statement Parser::parseStatement()
...
+	case Token::Address: // opcode
 	default:
 		break;
 	}
 	// Options left:
 	// Simple instruction (might turn into functional),
 	// literal,
 	// identifier (might turn into label or functional assignment)
-	Statement statement(parseElementaryOperation());
+	Statement statement(parseElementaryOperation(false));
```
Comment (axic) [inline libsolidity/inlineasm/AsmParser.cpp:95]: Making the `false` explicit here.

**Comment by axic** [comment]
I think I'll remove the analyzer tests because that is unfinished for Julia.

File: libsolidity/inlineasm/AsmPrinter.h [hunk 3]
```diff
@@ -45,6 +45,8 @@ struct Block;
 class AsmPrinter: public boost::static_visitor<std::string>
 {
 public:
+	AsmPrinter(bool _julia = false): m_julia(_julia) {}
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmPrinter.h]: Please add `explicit`.

File: libsolidity/inlineasm/AsmParser.cpp [hunk 4]
```diff
@@ -209,7 +214,11 @@ assembly::Statement Parser::parseElementaryOperation(bool _onlySinglePusher)
 		break;
 	}
 	default:
-		fatalParserError("Expected elementary inline assembly operation.");
+		fatalParserError(
+			m_julia ?
+			"Expected literal or identifier." :
+			"Expected elementary inline assembly operation."
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmParser.cpp]: Can you make the error messages a bit more uniform, at least per component? Above, you added an error message that ends with `expected`, here the message starts with `expected`.

**Review by chriseth — APPROVED** [review]
Fine to merge after the requested small changes.

File: libsolidity/inlineasm/AsmParser.cpp (hunk 4 above)
Comment (axic) [reply libsolidity/inlineasm/AsmParser.cpp]: So which order do we want, expected in front or back? It is kind of random at the moment.

**Pull request merged**